src/DataForm/Field/Autocomplete.php

```diff
@@ -153,26 +153,44 @@ public function build()
                     var blod_{$this->name} = new Bloodhound({
                         datumTokenizer: Bloodhound.tokenizers.obj.whitespace('auto_{$this->name}'),
                         queryTokenizer: Bloodhound.tokenizers.whitespace,
-                        remote: '{$this->remote}?q=%QUERY'
+                        remote: {
+                            url: '{$this->remote}?q=%QUERY',
+                            ajax: {
+                                complete: function(response){
+                                    response.responseJSON.forEach(function (item) {
+                                        blod_{$this->name}.valueCache[item.{$this->record_label}] = item.{$this->record_id};
+                                    });
+                                }
+                            }
+                        }
                     });
+                    blod_{$this->name}.valueCache = {};
                     blod_{$this->name}.initialize();
 
-                    $('#th_{$this->name} .typeahead').typeahead(null, {
-                        name: '{$this->name}',
-                        displayKey: '{$this->record_label}',
+                    $('#th_{$this->name} .typeahead').typeahead({
                         highlight: true,
                         minLength: {$this->min_chars},
+                    }, {
+                        displayKey: '{$this->record_label}',
+                        name: '{$this->name}',
                         source: blod_{$this->name}.ttAdapter(),
                         templates: {
                             suggestion: Handlebars.compile('{{{$this->record_label}}}')
                         }
                     }).on("typeahead:selected typeahead:autocompleted",
-                        function (e,data) { $('#{$this->name}').val(data.{$this->record_id});
-
+                        function (e,data) {
+                            $('#{$this->name}').val(data.{$this->record_id});
                     }).on("typeahead:closed",
                         function (e,data) {
-                            if ($(this).val() == '') {
-                                $('#{$this->name}').val('');
+                            if ('{$this->must_match}') {
+                                if ( !($.trim($(this).val()) in blod_{$this->name}.valueCache) ) {
+                                    $('#{$this->name}').val('');
```

Comment:
@zofe I haven't find really good way to notify the user, so simply deleted the value inputed. Do you have any suggestions?